rust/actyx/ax-aql/src/language/aql.pest

```diff
@@ -7,23 +7,68 @@ main_timestamp = @{ SOI ~ isodate ~ EOI }
 
 // pragmas
 pragma = ${ "PRAGMA" ~ WHITE_SPACE+ ~ feature_word ~ ( !NEWLINE ~ WHITE_SPACE )* ~ ( pragma_multi | pragma_single ) }
-pragma_multi = _{ NEWLINE ~ pragma_multi_text ~ NEWLINE ~ "ENDPRAGMA" ~ NEWLINE }
-pragma_multi_text = ${ ( !( NEWLINE ~ "ENDPRAGMA" ~ NEWLINE ) ~ ANY )* }
+pragma_multi = _{ NEWLINE ~ pragma_multi_text ~ NEWLINE ~ pragma_end ~ NEWLINE }
+pragma_multi_text = ${ ( !( NEWLINE ~ pragma_end ~ NEWLINE ) ~ ANY )* }
+pragma_end = { "ENDPRAGMA" }
 pragma_single = _{ ":=" ~ ( !NEWLINE ~ WHITE_SPACE )* ~ pragma_single_text ~ NEWLINE }
 pragma_single_text = ${ ( !NEWLINE ~ ANY )* }
 
+features = { "FEATURES(" ~ feature_word* ~ ")" }
+feature_word = @{ ( ASCII_ALPHANUMERIC | "ø" )+ }
+
+// types
+type = { type_atom ~ type_post* ~ ( type_infix ~ type_atom ~ type_post* )* }
+type_null = { "NULL" }
+type_bool = { "BOOLEAN" }
+type_number = { "NUMBER" }
+type_timestamp = { "TIMESTAMP" }
+type_string = { "STRING" }
+type_universal = { "UNIVERSAL" }
```

Comment:
Yes, this is the ⊤ type, which is called `Any` in Scala, `unknown` in Typescript, `Object` in Java, … Using `ANY` felt wrong because `any` in Typescript is probably most well-known and means something else. And `UNKNOWN` doesn’t convey the right connotations as well, because it implies that knowledge is missing.